More events. Remove unnecessary bool return values

yondonfu opened the pull request:
- Added more events that I thought might be useful for a client interacting with the contracts
- Removed unnecessary bool returns values for state mutating functions. Since our convention is to fail hard and fail early by reverting transactions upon any type of error, bool return values are never useful. There are some cases that a bool return value might be useful even if we're reverting, but I don't believe there are any cases of that in our code right now.
- Removed the constructor and fallback function from LivepeerToken because they shouldn't be needed - no ETH can be sent to the contract if the fallback function is not payable and the fallback function is non-payable by default

Current list of events (excluding LivepeerVerifier and LivepeerToken):

ParameterUpdate
SetContract
SetController
TranscoderUpdate
TranscoderEvicted
TranscoderResigned
Bond
Unbond
Reward
NewRound
NewJob
NewClaim
Verify
DistributeFees
ReceivedVerification
NewInflation
SetCurrentRewardTokens

Let me know if there are any additional events in particular that anyone thinks might be particularly helpful. Maybe @jozanza has some thoughts relevant for the SDK?

Closes #31

dob commented:
What about events for 

* WIthdrawing funds
* Failing Verification
* Slashing

dob commented on contracts/Migrations.sol:
```diff
@@ -10,15 +10,15 @@ contract Migrations {
     }
 
-    function Migrations() {
+    function Migrations() public {
```
These methods that are marked `public`...they're `public` by default and we're just adding this to be explicit, right? Good idea.

yondonfu replied:
Yep, the compiler also emits warnings if you don't add explicit visibility declarations - a non-functionality related todo is to do clean up any compiler warnings and make the CI fail if we have any warnings

dob commented on contracts/bonding/BondingManager.sol:
```diff
@@ -282,7 +292,7 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         }
 
-        return true;
+        Bond(_to, msg.sender, currentRound);
```
Are we including `currentRound` in all the events for convenience and searchability on the frontend? Presumably the client would know which block or round it occured in based upon the event metadata itself, right...but this would just allow us to filter or easily view a timeline?

yondonfu replied:
Yeah good point - the current round can always be computed from the block number during which an event is received

yondonfu commented:
Removed round values from events as they can always be recalculated using the block number associated with an event. Also added `Withdraw`, `Deposit` and verification events with more specific naming (`PassedVerification` and `FailedVerification`)
